native/src/main/java/io/ballerina/stdlib/mqtt/caller/CallerThreadFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.mqtt.caller;
+
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * A {@link ThreadFactory} object that creates new threads on demand for MQTT caller network actions.
+ */
+class CallerThreadFactory implements ThreadFactory {
+    @Override
+    public Thread newThread(Runnable runnable) {
+        Thread jmsConsumerThread = new Thread(runnable);
+        jmsConsumerThread.setName("bal-mqtt-caller-network-thread");
+        return jmsConsumerThread;
```

Comment:
```suggestion
        Thread mqttClientThread = new Thread(runnable);
        mqttClientThread.setName("bal-mqtt-caller-network-thread");
        return mqttClientThread;
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-mqtt/pull/9/commits/56d4c80f1818864c9e0d79f5702f9ccc115eaf87